packages/deamon/src/pid.rs

```diff
@@ -0,0 +1,189 @@
+use super::config::ProjectConfig;
+use anyhow::Result;
+use log::{debug, error, info};
+use nix::sys::signal::{self, Signal};
+use nix::unistd;
+use std::sync::Arc;
+use std::{collections::BTreeMap, process::Command};
+use tokio::{
+    sync::Mutex,
+    time::{timeout, Duration},
+};
+
+#[derive(Debug, PartialEq, Eq)]
+pub struct ProjectInfo {
+    pub config: ProjectConfig,
+    pub pids: Vec<u32>,
+}
+
+impl ProjectInfo {
+    pub fn kill_pids(&mut self) {
+        for pid in self.pids.iter() {
+            info!(
+                "{:?} be killing, project path {}",
+                pid,
+                self.config.get_project_path()
+            );
+            match signal::kill(unistd::Pid::from_raw(pid.clone() as i32), Signal::SIGKILL) {
+                Ok(_) => {
+                    info!(
+                        "{:?} killed, project path {}",
+                        pid,
+                        self.config.get_project_path()
+                    );
+                }
+                Err(err) => {
+                    error!("Failed to kill process with PID {}. Error: {:?}", pid, err);
```

Comment:
Errors should be propagated upwards instead of being logged directly.

Reply:
when unioufs is killed, an error will be reported if nfsv4 is killed again, but the error should not be reported upward to prevent restarting.

Reply:
NFV4 is a subprocess of UnionFS with its own monitoring mechanism. Shouldn't we exclude it from our monitoring scope?

Reply:
done